### ISSUE TITLE
Support deep keys

### ISSUE THREAD

**Issue by grosser** [airbrake/airbrake-ruby#137]
rails filter_parameters allows `foo.bar` meaning `foo[bar]` 
which is not compatible with airbrakes blacklist filter
so please either:
 - support it (might be tricky, but a useful filter)
 - change airbrake rails docs to point this out

**Comment by kyrylo** [comment]
Thanks for filing this! I think we already support that on notifier level (if you filter just `foo`, it'll be filtered everywhere, even in:

```rb
{
  not_foo: {
    not_foo_again: {
      foo: 'will be filtered'
    }
  }
}
```

But in this regard our API is probably indeed incompatible. I guess it's better to document this and ignore `bar` in `foo[bar]`. That said, it's possible to ignore both, `foo` and `bar` (globally, like we do as of now), but it might be risky.

**Comment by kyrylo** [comment]
Made some important edits to my original message. Please reload this page if it's open :)

[this post will be auto-destructed]

**Comment by grosser** [comment]
doing this atm ... just a evil trap that the docs say 'put filter into blacklist' and then they do not work the same way ... which ends up leaking secrets into airbrake ... not fun :(

**Comment by kyrylo** [comment]
Sorry about that. I was simply unaware of the `foo.bar` syntax. It seems like it's a Rails 5 feature.

**Comment by grosser** [comment]
would you accept a PR to add this feature ? ... I think it's pretty useful :)

**Comment by kyrylo** [comment]
Depends on the implementation. If there's a neat way to implement this, then why not? However our current policy is to filter out every matching key, no matter how deep it is. We also support whitelists. This might make the implementation harder. Without special cases the code for truncation is nice, fast and tidy.

TL;DR Go for it, but it might be harder than you think :P 
P.S. But probably not too hard

**Comment by grosser** [comment]
took a look ... would mean rewriting filtering logic ... and make it a bunch slower ... so not doing it ...
